### PR TITLE
[DCOS-49117] spark-build-image-refactoring

### PR DESCRIPTION
**Pull request by viivek46**
## What changes were proposed in this pull request?

Resolves [DCOS-49117](https://jira.mesosphere.com/browse/DCOS-49117)

- Switched base image from version to latest SHA tag,
- Installing R from Ubuntu package repository and removed build R from sources,
- Apache license added
- cleanup unused/irrelevant entries

## How were these changes tested?

Integration tests from this repo

### CODE REVIEW

**Comment by viivek46** [comment]
@akirillov @samvantran Could you please review this PR.

File: Dockerfile [hunk 1]
```diff
@@ -26,7 +48,6 @@ RUN add-apt-repository -y ppa:longsleep/golang-backports && apt-get update && ap
 # AWS CLI for uploading build artifacts
 RUN pip install awscli
```
Comment (akirillov) [inline Dockerfile]: should become pip3

File: Dockerfile [hunk 2]
```diff
@@ -16,7 +38,7 @@ RUN apt-get update && apt-get install -y \
     python3-dev \
     python3-pip \
     python3-venv \
-    python-software-properties \
+    r-base \
     software-properties-common \
     wget \
     zip && \
```
Comment (akirillov) [inline Dockerfile]: it looks like this package list has been copied from somewhere and we don't need the majority of these packages. E.g. `bc` is a calculator. I suggest shrinking it down to:
```
build-essential \
curl \
groff \
python3 \
python3-pip \
r-base \
software-properties-common
```

Note, `groff` is required by awscli

**Comment by akirillov** [comment]
echoing @samvantran with regards to PR body, you can check this PR as an example https://github.com/mesosphere/spark-build/pull/485

File: Dockerfile (hunk 2 above)
Comment (viivek46) [reply Dockerfile]: updated docker file as suggested.

File: Dockerfile (hunk 1 above)
Comment (viivek46) [reply Dockerfile]: updated docker file as suggested.

File: Dockerfile [hunk 3]
```diff
@@ -26,7 +48,6 @@ RUN add-apt-repository -y ppa:longsleep/golang-backports && apt-get update && ap
 # AWS CLI for uploading build artifacts
 RUN pip install awscli
 # Install dcos-launch to create clusters for integration testing
-RUN apt-get install -y python3-venv
 RUN wget https://downloads.dcos.io/dcos-launch/bin/linux/dcos-launch -O /usr/bin/dcos-launch
 RUN chmod +x /usr/bin/dcos-launch
```
[the remark this answers is not in the document]
Comment (viivek46) [reply Dockerfile]: updated docker file as suggested.

File: Dockerfile [hunk 4]
```diff
@@ -35,17 +56,6 @@ ENV LANG=C.UTF-8
 # use an arbitrary path for temporary build artifacts
 ENV GOPATH=/go-tmp
```
[the remark this answers is not in the document]
Comment (viivek46) [reply Dockerfile]: updated docker file as suggested.

File: Dockerfile [hunk 5]
```diff
@@ -1,55 +1,53 @@
...
-    git \
-    jq \
-    libbz2-dev \
-    liblzma-dev \
-    libcurl4-openssl-dev \
-    libpcre++-dev \
-    libssl-dev \
+    openjdk-8-jdk \
+    groff \
     python-pip \
```
Comment (akirillov) [inline Dockerfile]: this doesn't seem to be needed as we're installing `python3-pip`

**Pull request merged**